### PR TITLE
Performance and IDEA autocomplete

### PR DESCRIPTION
**Pull request by d954mas**
1)add autocomplete for IDEA
2) increase speed of fill ~ 10x
2) increase speed of fill_line ~ 3x

### CODE REVIEW

File: drawpixels/src/drawpixels.cpp [hunk 1]
```diff
@@ -62,9 +62,33 @@ static void fill_line(int fromx, int tox, int y, int r, int g, int b, int a){
...
+  int width = (end - start+buffer_info.channels);
+  for(int i = start; i < end; i +=line_size) {
+    if(width >= line_size){
+      memcpy(&buffer_info.bytes[i], line, line_size);
+    }else{
+      memcpy(&buffer_info.bytes[i], line, width);
+    }
+    width = width - line_size;
   }
+  delete line;
```
Comment (AGulev) [inline drawpixels/src/drawpixels.cpp]: Compiler warning:

I think here should be 
delete[] line;

File: drawpixels/src/drawpixels.cpp [hunk 2]
```diff
@@ -336,23 +360,28 @@ static int fill_texture(lua_State* L) {
...
     if (buffer_info.channels == 4) {
-      buffer_info.bytes[i + 3] = a;
+      line[i + 3] = a;
     }
   }
-
+  for(int i = 0; i < buffer_info.src_size; i +=line_size) {
+    memcpy(&buffer_info.bytes[i], line, line_size);
+  }
+  delete line;
```
Comment (AGulev) [inline drawpixels/src/drawpixels.cpp]: Compiler warning:


I think here should be 
delete[] line;

File: drawpixels/src/drawpixels.cpp [hunk 3]
```diff
@@ -62,9 +62,33 @@ static void fill_line(int fromx, int tox, int y, int r, int g, int b, int a){
     fromx = tox;
     tox = temp;
   }
-  for (int x = fromx; x <= tox; x++) {
-    putpixel(x, y, r, g, b, a);
+  fromx = max(0, fromx);
+  tox = min(buffer_info.width-1, tox);
```
Comment (AGulev) [inline drawpixels/src/drawpixels.cpp]: What abot fmax/fmin, or maybe our own defines? https://stackoverflow.com/questions/7035023/stdmax-expected-an-identifier?answertab=votes#tab-top

File: drawpixels/src/drawpixels.cpp (hunk 1 above)
Comment (d954mas) [reply drawpixels/src/drawpixels.cpp]: Yes, you are right.

File: drawpixels/src/drawpixels.cpp (hunk 3 above)
Comment (d954mas) [reply drawpixels/src/drawpixels.cpp]: Change to fmax.

**Pull request merged**